include/range/v3/utility/swap.hpp

```diff
@@ -42,10 +53,26 @@ namespace ranges
         /// \cond
         namespace adl_swap_detail
         {
-            using std::swap;
+
+            // The following implementation of swap is adapted from libc++:
+            template <class T>
+            RANGES_CXX14_CONSTEXPR
+            typename std::enable_if<
+                std::is_move_constructible<T>::value &&
+                std::is_move_assignable<T>::value
+             >::type
+            swap(T& x, T& y)
+            noexcept(std::is_nothrow_move_constructible<T>::value &&
+                     std::is_nothrow_move_assignable<T>::value)
+            {
+                T t(std::move(x));
+                x = std::move(y);
+                y = std::move(t);
+            }
```

Comment:
...And it seems to me that if you try to call `swap` on a type that happens to have `std` as an associated namespace, it's going to find the one in `std` anyway. This overload creates an ambiguity error, does it not?